### DISCUSSION TITLE
HelpRequest : UseFieldArray remove problem

### DISCUSSION THREAD

**Discussion by sunilbittu** [react-hook-form/react-hook-form#2495] [Q&A] [answered]
Hi Please help me to resolve the remove issue while using react-hook-form useFieldArray

https://codesandbox.io/s/react-hook-form-usefieldarray-36pju?file=/src/index.js

Please tell me where i'm doing wrong

**Comment by bluebill1049** [comment, marked as answer]
have a read this section: https://react-hook-form.com/api#useFieldArray
here is a codesandbox example to follow: https://codesandbox.io/s/react-hook-form-usefieldarray-vy8fv

**Reply by sunilbittu** [reply]
Hi Bill, i read the documentation and implemented this piece of code, values are adding properly onSubmit 
Can u help me to achieve particular pickup deletion in my code, remove(index) is what i'm trying to.

`<div>
                  <button onClick={() => remove(index)}>Remove</button>
                </div>`

**Reply by bluebill1049** [reply]
did you give `button` `type="button"`?

**Reply by sunilbittu** [reply]
I did that sir, getting this error on remove 

data.slice is not a function or its return value is not iterable

**Reply by bluebill1049** [reply]
couple things: 

1. ref={register()}
2. key={item.id}
3. missing defaultValue

https://codesandbox.io/s/react-hook-form-usefieldarray-tc6hw?file=/src/index.js

it's important to read the doc, all those dot points are covered in the doc. please give it another read, especially this section: 

<img width="1003" alt="Screen Shot 2020-08-06 at 7 55 11 pm" src="https://user-images.githubusercontent.com/10513364/89518781-bf666a00-d81e-11ea-8997-37809dfefa5b.png">

**Reply by sunilbittu** [reply]
Thanks Bill, i think i have messed up something , let me read properly again thank you

**Reply by sunilbittu** [reply]
Thanks bill, able to achieve the desired functionality.
